### PR TITLE
feat(tekton)!: enable tekton related features when the tekton annotation is present

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://github.com/janus-idp/backstage-plugins/issues/700

### CODE REVIEW

File: plugins/tekton/README.md [hunk 1]
```diff
@@ -76,13 +76,13 @@ The Tekton plugin enables you to visualize the `PipelineRun` resources available
...
 
-- The following annotation is added to the `catalog-info.yaml` file of entity to view the latest `PipelineRun` in the CI/CD tab of the application:
+- The following annotation is added to the `catalog-info.yaml` file of the entity to enable the Tekton related features in Backstage. The value of the annotation identifies the name of the application:
 
   ```yaml
   annotations:
     ...
 
-    janus-idp.io/tekton-enabled : 'true'
+    janus-idp.io/tekton : <APP_NAME>
```
Comment (gorkem) [inline plugins/tekton/README.md:85]: It feels like this needs more explanation of where the <APP_NAME> is coming from.

**Comment by tumido** [comment]
@debsmita1 I wonder, is this a breaking change? Because it seems like it. If so, please rename the PR to `feat(tekton)!:...`

File: plugins/tekton/README.md (hunk 1 above)
Comment (debsmita1) [reply plugins/tekton/README.md]: updated this to BACKSTAGE_ENTITY_NAME

**Review by gashcrumb — APPROVED** [review]
tested locally, LGTM :+1:

File: plugins/tekton/README.md [hunk 2]
```diff
@@ -76,13 +76,13 @@ The Tekton plugin enables you to visualize the `PipelineRun` resources available
...
 
-- The following annotation is added to the `catalog-info.yaml` file of entity to view the latest `PipelineRun` in the CI/CD tab of the application:
+- The following annotation is added to the `catalog-info.yaml` file of the entity to enable the Tekton related features in Backstage. The value of the annotation identifies the name of the Backstage entity:
 
   ```yaml
   annotations:
     ...
 
-    janus-idp.io/tekton-enabled : 'true'
+    janus-idp.io/tekton : <BACKSTAGE_ENTITY_NAME>
```
Comment (invincibleJai) [inline plugins/tekton/README.md:85]: currently, this annotations just help to detect if Tekton needs to be enabled or not 

can we have something as below

```
janus-idp.io/tekton/porject-slug: <value>
janus-idp.io/tekton/app-name : <value>
```


@christophe-f Let us know your thoughts.

Comment (christophe-f) [reply plugins/tekton/README.md:85]: It's fine like this.

**Pull request merged**